src/equations/bbm_1d.jl

```diff
@@ -0,0 +1,241 @@
+@doc raw"""
+    BBMEquation1D(; bathymetry_type = bathymetry_flat,
+                  gravity_constant, D = 1.0, eta0 = 0.0, split_form = true)
```

Comment:
Since we only support `bathymetry_flat` right now - and I am not sure whether it makes sense to generalize this as discussed earlier - I wonder whether we should drop the `bathymetry_type` here?

Reply:
[b40b9b4](https://github.com/JoshuaLampert/DispersiveShallowWater.jl/pull/150/commits/b40b9b4ba8e8c228ca703a7e990b402f5295e16e)